### ISSUE TITLE
Problem with live youtube stream

### ISSUE THREAD

**Issue by refigh** [mps-youtube/pafy#239]
Hi
I was able to use this tool to play YouTube video in Python 3+OpenCV 4.1. However, when I use URL of a live YouTube I get below error:

  File "/usr/local/lib/python3.6/dist-packages/pafy/pafy.py", line 124, in new
    return Pafy(url, basic, gdata, size, callback, ydl_opts)
  File "/usr/local/lib/python3.6/dist-packages/pafy/backend_youtube_dl.py", line 29, in __init__
    super(YtdlPafy, self).__init__(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pafy/backend_shared.py", line 96, in __init__
    self._fetch_basic()
  File "/usr/local/lib/python3.6/dist-packages/pafy/backend_youtube_dl.py", line 47, in _fetch_basic
    self._author = self._ydl_info['uploader']
KeyError: 'uploader'

tested live vide: "https://www.youtube.com/watch?v=YlksXBZbx8Y"
moreover, this command also generates same error:
sudo ytdl -b "https://www.youtube.com/watch?v=YlksXBZbx8Y"

Is there any support for live video?

**Comment by abhiTronix** [comment]
**Test this:** https://github.com/abhiTronix/vidgear/issues/16#issuecomment-495456217 and drop a :star: if helped. Goodluck!

**Comment by dmarraud** [comment]
Hello, I am trying to use pafy and opencv to access youtube live streams. The first two or three seconds are grabbed correctly, then it freezes and next frames are grabbed after tens of seconds then freeze again, and so on. Beside this problem I would have expected a capture that takes the latest available image in the live stream whatever the grabbing rate, whereas current behavior seems to show that all frames in the stream have to be grabbed sequentially? Any ideas? Thanks!

**Comment by refigh** [comment]
frames in mpeg/h264 standard format stream are dependent on each other (they are not like independent images). So, if you lose one frame, next frame can not be decoded. 
I had this problem too, not only with live youtube, but also with IP cameras, onan ARM board. I think the problem is that opencv does not use gpu (or cpu's internal decoder) to decode frames. I temporary solved my issue by reducing the FPS of my IP camera, but generally 
 you need to find another way (like read stream with VLC, gstream,..etc, and somehow pass the result to opencv)
see this : https://github.com/opencv/opencv/issues/15545

**Comment by dmarraud** [comment]
Thanks for your answer! What is Strange is that it works perfectly well with non live YouTube videos.

**Comment by refigh** [comment]
because offline youtube is like file, you read offline, by your own speed, no packet is lost. but in online youtube, or IP-camera, you need to decode this frame, before losing next frame.

**Comment by dmarraud** [comment]
Yes it makes sense. Thanks. Thus, any other link to share that would show how to access decoded live streaming frames in opencv ? I searched different forums without success so far...

**Comment by refigh** [comment]
what is your platform? do you work on an ARM board, like Raspberry?

**Comment by dmarraud** [comment]
No. Standard desktop under Linux or Windows.Le 22 sept. 2019 20:39, Refigh <notifications@github.com> a écrit :what is your platform? do you work on an ARM board, like Raspberry?

**Comment by refigh** [comment]
As I wrote, I recently learnt that opencv does not use GPU/internal decoder/encoder,... and you need to find a way to use them.
As far as I know, gstreamer and VLC use GPUs. I read somewhere (but will test this week myself) to use gstreamer and its python plugin. It means use a linux/windows tool like gstreamer to get the frames and decode them fast, then pass them to python+opencv with help of special plugin.
